docs/App/Header.js

```diff
@@ -117,10 +117,11 @@ class Header extends Component<HeaderProps, HeaderState> {
   componentDidMount() {
     this.getStarCount();
   }
-  UNSAFE_componentWillReceiveProps({ location }: HeaderProps) {
+  componentDidUpdate(prevProps: HeaderProps) {
     const valid = ['/', '/home'];
-    const shouldCollapse = !valid.includes(this.props.location.pathname);
-    if (location.pathname !== this.props.location.pathname && shouldCollapse) {
+    const shouldCollapse = !valid.includes(prevProps.location.pathname);
+
+    if (prevProps.location.pathname !== this.props.location.pathname && shouldCollapse) {
       this.toggleCollapse();
     }
   }
```

Comment:
Doing this on update means that the first time you click away from the home page, the transition won't work (which is actually true of the status quo as well).

Instead I would suggest that we call `this.setState({ contentHeight: this.content.scrollHeight });` immediately in `componentDidMount` and get rid of `toggleCollapse` altogether.